### PR TITLE
fix committed sequence twice

### PR DESCRIPTION
**Pull request by pfi79**
please check these changes very carefully

### CODE REVIEW

File: internal/bft/controller.go [hunk 1]
```diff
@@ -535,12 +536,11 @@ func (c *Controller) run() {
 func (c *Controller) decide(d decision) {
 	c.Logger.Debugf("Delivering to app from Controller decide the last decision proposal")
 	begin := time.Now()
-	reconfig := c.Application.Deliver(d.proposal, d.signatures)
+	reconfig := c.Deliver.Deliver(d.proposal, d.signatures)
 	c.MetricsView.LatencyBatchSave.Observe(time.Since(begin).Seconds())
```
Comment (HagarMeir) [inline internal/bft/controller.go]: Maybe move the metrics to be only on the application deliver

File: internal/bft/controller.go [hunk 2]
```diff
@@ -954,7 +954,7 @@ func (med *MutuallyExclusiveDeliver) Deliver(proposal types.Proposal, signature
 	// do not proceed to commit the proposal, but instead invoke a sync and update the checkpoint once more
 	// to match the sync result.
 	latest := med.C.latestSeq()
-	if latest >= pendingProposalMetadata.LatestSequence {
+	if latest != 0 && latest >= pendingProposalMetadata.LatestSequence {
```
Comment (HagarMeir) [inline internal/bft/controller.go:955]: Why do you check if `latest != 0`

File: internal/bft/controller.go (hunk 1 above)
Comment (pfi79) [reply internal/bft/controller.go]: Yes you're right, I'll correct that.

If you don't mind, I'll correct this one in another pr

File: internal/bft/controller.go (hunk 2 above)
Comment (pfi79) [reply internal/bft/controller.go:955]: When latest == 0 it is both "sequence is 0" and "no sequence". 
I did this to go from "no sequence" to "first sequence with index 0" state.

File: internal/bft/controller.go (hunk 1 above)
Comment (pfi79) [reply internal/bft/controller.go]: I corrected that.

**Review by C0rWin — APPROVED** [review]

**Pull request merged**